src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java

```diff
@@ -330,6 +354,84 @@ public Response getIterationDetails(
         return Response.ok(gtfsRtFeedIterationModel).build();
     }
 
+    // Returns past session details for a particular clientId.
+    @GET
+    @Path("/{clientId : .*}/pastSessions")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getSessionData(
+            @PathParam("clientId") String clientId) {
+
+        if (clientId.isEmpty()) {
+            String uuid = UUID.randomUUID().toString();
+            JsonObject json = new JsonObject();
+            json.addProperty("clientId", uuid);
+            return Response.ok(json.toString()).build();
+        }
+
+        List<SessionModel> sessionModelList;
+        Session session = GTFSDB.initSessionBeginTrans();
+        long timeDiff;
+        int rowId = 0;
+
+        sessionModelList = session.createQuery(" FROM SessionModel" +
+                " WHERE userId = '" + clientId + "'").list();
+
+        Iterator iterator = sessionModelList.listIterator();
+        SessionModel eachSessionModel;
+        DateFormat timeFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
+        while(iterator.hasNext()) {
+            eachSessionModel = (SessionModel) iterator.next();
+
+            eachSessionModel.setRowId(++rowId);
```

Comment:
Could you please split this into two separate statements - `eachSessionModel.setRowId(rowId);` here, and `rowId++` after it?  As a style I prefer not to use `++` before the variable use.